src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -452,9 +452,24 @@ public static Map<String, DetailNode> getPropertiesJavadocs(Set<String> properti
         final Map<String, DetailNode> unmodifiableJavadocs =
                 ClassAndPropertiesSettersJavadocScraper.getJavadocsForModuleOrProperty();
         final Map<String, DetailNode> javadocs = new LinkedHashMap<>(unmodifiableJavadocs);
-        properties.forEach(property -> {
-            javadocs.putIfAbsent(property, SUPER_CLASS_PROPERTIES_JAVADOCS.get(property));
-        });
+
+        for (String property : properties) {
+            final DetailNode superClassPropertyJavadoc = SUPER_CLASS_PROPERTIES_JAVADOCS
+                    .get(property);
+            if (superClassPropertyJavadoc != null) {
+                javadocs.putIfAbsent(property, superClassPropertyJavadoc);
+            }
```

Comment:
why we do this put in a loop ? can we do it outside ones

Reply:
Like in a different method?

Reply:
my bad, I misread a code

Reply:
please make it code formatted like:
```
final DetailNode superClassPropertyJavadoc = 
   SUPER_CLASS_PROPERTIES_JAVADOCS.get(property);

```

Reply:
confusion happens while reading this code due to combining validation and data-enrichment in same loop.
please split into two loops:
```
properties.forEach(property -> {
            javadocs.putIfAbsent(property, SUPER_CLASS_PROPERTIES_JAVADOCS.get(property));
        });
```
and 
```
properties.forEach(property -> {
           .... validation isPropertySetterJavadocFound
        });
```